internal/command/commands/service.go

```diff
@@ -40,18 +40,31 @@ func (s *Service) Run(args []string) int {
 	}
 
 	if s.Create {
-
-		emptyParameters := emptyParameters(map[string]string{"--name": *serviceName, "--version": *serviceVersion})
-		if len(emptyParameters) == 0 {
+		emptyCreateParameters := emptyParameters(map[string]string{"--name": *serviceName, "--version": *serviceVersion})
+		if len(emptyCreateParameters) == 0 {
 			serviceClient.RebuildServiceStream(*serviceName, *serviceVersion)
 			s.Logger.Output("Command to check status of images")
 			s.Logger.ItalicEmphasize(fmt.Sprintf("odin status service --name %s --version %s", *serviceName, *serviceVersion))
 			return 0
 		}
 
-		if len(*filePath) == 0 {
+		if len(*filePath) > 0 {
+			emptyFileParameters := emptyParameters(map[string]string{"--name": *serviceName, "--version": *serviceVersion})
+			split := strings.Split(emptyFileParameters, ",")
+			if len(split) < 2 {
+				s.Logger.Error("--name and --version should not be provided when --file is provided.")
+				return 1
+			}
+		} else {
 			*filePath = "service.json"
+
+			if (len(*serviceName) != 0 && len(*serviceVersion) == 0) ||
```

Comment:
What if both name and version are not given.
We can simplify condition to `len(*serviceVersion) == 0 || len(*serviceName) == 0`

Reply:
updated.